queue/queue.js

```diff
@@ -1,48 +1,50 @@
 
-function Queue(maxCapacity) {
-    this.maxCapacity = maxCapacity;
-    this.queue = {}
-    this.startIndex = 0; //start from this position
-    this.endIndex = 0; //Nothing to remove
-}
+class Queue {
+    constructor (maxCapacity = 10) {
+        this.maxCapacity = maxCapacity;
+        this.queue = {}
+        this.startIndex = 0; //start from this position
+        this.endIndex = 0; //Nothing to remove
+    }
 
-Queue.prototype.enqueue = function(value) {
+    enqueue(value) {
 
-    if( this.count() < this.maxCapacity)
-    {
-        this.queue[ this.endIndex++ ] = value;
+        if( this.count() < this.maxCapacity)
+        {
+            this.queue[ this.endIndex++ ] = value;
+        }
     }
-}
 
-Queue.prototype.dequeue = function() {
+    dequeue() {
 
-    if (this.count() > 0)
-    {
-        var valueDeleted = this.queue[ this.startIndex ];
-        delete this.queue[ this.startIndex++ ];
-        return valueDeleted;
+        if (this.count() > 0)
+        {
+            let valueDeleted = this.queue[ this.startIndex ];
+            delete this.queue[ this.startIndex++ ];
+            return valueDeleted;
+        }
     }
-}
 
-Queue.prototype.peek = function() {
-    return (this.count() > 0)? this.queue[this.startIndex]:undefined;
-}
+    peek() {
+        return (this.count() > 0)? this.queue[this.startIndex]:undefined;
+    }
 
-Queue.prototype.count = function() {
```

Comment:
count can be a get property https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get